base/src/error.rs

```diff
@@ -86,3 +94,21 @@ impl From<cid::Error> for BlockError {
         Self::Cid(err)
     }
 }
+
+impl<T: std::fmt::Debug> From<RwLockReadGuard<'_, T>> for BlockError {
```

Comment:
I did not wanted to complicate `BlockError` by introducing parameters. I could not figure out a better way to do this, please help 😭

Reply:
@dvc94ch This is no longer will be a problem as store will go back to being async, but would you mind still commenting what would have being a right way to go about it ?

Reply:
The correct way to go about it would have been:
 ```rust
impl From<std::sync::PoisonError> for BlockError {
    fn from(error: std::sync::PoisonError) -> Self {
        Self::PoisonError(error)
    }
}
```
and the three cases would have collapsed into one if I'm not mistaken.